src/MSBuild/XMake.cs

```diff
@@ -2570,6 +2570,11 @@ private static bool ProcessTerminalLoggerConfiguration(CommandLineSwitches comma
 
             static bool DoesEnvironmentSupportTerminalLogger()
             {
+                if (Environment.GetEnvironmentVariable("MSBUILDTESTINGFORCETERMINALLOGGERUSE") == "1")
```

Comment:
Is this necessary? Couldn't this be achieved by a standard switch or mechanism?
I'm not big fan of modifying production code just for test.

cc: @rokonec

Reply:
Oh, I looked closer on the switch and there is a way to force it on already indeed. Used it.